ospf6d/ospf6_top.c

```diff
@@ -878,6 +880,75 @@ DEFUN (no_ospf6_stub_router_admin,
 	return CMD_SUCCESS;
 }
 
+/* Restart OSPF SPF algorithm*/
+static void ospf6_restart_spf(struct ospf6 *ospf6)
+{
+	/* Handling inter area and intra area routes*/
+	ospf6_route_remove_all(ospf6->route_table);
+	ospf6_route_remove_all(ospf6->brouter_table);
+	ospf6_route_remove_all(ospf6->external_table);
+
+	/* Trigger SPF */
+	ospf6_spf_schedule(ospf6, OSPF6_SPF_FLAGS_CONFIG_CHANGE);
+}
+
+/* Set the max paths */
+static void ospf6_maxpath_set(struct ospf6 *ospf6, uint16_t paths)
+{
+	if (ospf6->max_multipath == paths)
+		return;
+
+	ospf6->max_multipath = paths;
+
+	/* Send deletion to zebra to delete all
+	 * ospf specific routes and reinitiate
+	 * SPF to reflect the new max multipath.
+	 */
+	ospf6_restart_spf(ospf6);
+}
+
+/* Ospf Maximum-paths config support */
+DEFUN(ospf6_max_multipath,
+      ospf6_max_multipath_cmd,
+      "maximum-paths " CMD_RANGE_STR(1, MULTIPATH_NUM),
+      "Max no of multiple paths for ECMP support\n"
+      "Number of paths\n")
+{
+	VTY_DECLVAR_CONTEXT(ospf6, ospf6);
+	int idx_number = 1;
+	int maximum_paths = strtol(argv[idx_number]->arg, NULL, 10);
+
+	ospf6_maxpath_set(ospf6, maximum_paths);
+
+	return CMD_SUCCESS;
+}
+
+DEFUN(no_ospf6_max_multipath,
+      no_ospf6_max_multipath_cmd,
+     "no maximum-paths [" CMD_RANGE_STR(1, MULTIPATH_NUM)"]",
+      NO_STR
+      "Max no of multiple paths for ECMP support\n"
+      "Number of paths\n")
+{
+	VTY_DECLVAR_CONTEXT(ospf6, ospf6);
+
+	int idx_number = 2;
+	int maximum_paths = MULTIPATH_NUM;
+
+	if (argc > 2) {
+		maximum_paths = strtol(argv[idx_number]->arg, NULL, 10);
+		/* If the user gives a value which is not configured, return
```

Comment:
I don't think we usually are this restrictive - usually if the user issues the 'no' form, we just return to the default. Is there some special situation where you think this extra checking is needed?

Reply:
If user does not provide any input i.e "no maximum-paths" is configured, it is allowed and it removes the configuration. The "no maximum-paths <x>" is provided so that in case user wants to delete the config by using copy paste from the running config, it will able to execute this operation. So If wrong value is given in the input, it is not fine. Hence I have added that check.

Reply:
as I said, we are not typically this restrictive with the 'no' form. unless there's some specific reason not to accept 'no', I think you should just ... accept it.

Reply:
Okay